src/System.Text.RegularExpressions/ref/System.Text.RegularExpressions.cs

```diff
@@ -62,9 +62,22 @@ public partial class MatchCollection : System.Collections.ICollection, System.Co
         void System.Collections.ICollection.CopyTo(System.Array array, int arrayIndex) { }
     }
     public delegate string MatchEvaluator(System.Text.RegularExpressions.Match match);
+    abstract public partial class RegexRunnerFactory
+    {
+        protected RegexRunnerFactory() { }
+        abstract protected internal RegexRunner CreateInstance();
+    }
     public partial class Regex
     {
         public static readonly System.TimeSpan InfiniteMatchTimeout;
+        protected internal string pattern;
+        protected internal RegexRunnerFactory factory;
+        protected internal RegexOptions roptions;
+        protected internal TimeSpan internalMatchTimeout;
+        protected internal int capsize;
+        protected internal string[] capslist;
+        protected System.Collections.IDictionary Caps { get; set; }
+        protected System.Collections.IDictionary CapNames { get; set; }
```

Comment:
For better or worse the Desktop framework isn't going anywhere anytime soon so we are trying to keep things as compatible as possible. I don't think there is a big enough reason to switch to generic interfaces here.